src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,50 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeFromAlive() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+  }
+
+  bite(animal) {
+    if (
+      animal instanceof Carnivore ||
+      (animal.hidden && animal instanceof Herbivore)
+    ) {
+      return;
+    }
+
+    animal.health -= 50;
```

Comment:
```suggestion
    if (!animal.hidden && animal instanceof Herbivore) {
      animal.health -= 50;
    }
```

We can simplify it a bit